Add module.md and update README with setup guide information (#5)
ballerina-platform/module-ballerinax-discord: README.md

## module-balerinax-discord
[Discord](https://support.discord.com/hc/en-us/articles/360045138571-Beginner-s-Guide-to-Discord) is a popular communication platform designed for creating communities and facilitating real-time messaging, voice, and video interactions over the internet.

> The `ballerina-platform/module-ballerinax-discord` package offers APIs to connect and interact with the [Discord API](https://discord.com/developers/docs/reference)

Reviewer: ```suggestion
The Ballerina Discord connector offers APIs to connect and interact with the [Discord REST API v10](https://discord.com/developers/docs/reference)
```

> ## module-balerinax-discord
> [Discord](https://support.discord.com/hc/en-us/articles/360045138571-Beginner-s-Guide-to-Discord) is a popular communication platform designed for creating communities and facilitating real-time messaging, voice, and video interactions over the internet.

Reviewer: ```suggestion
## Overview 

[Discord](https://discord.com/) is a popular communication platform designed for creating communities and facilitating real-time messaging, voice, and video interactions over the internet.
```

The ``Ballerina Discord connector`` offers APIs to connect and interact with the [Discord REST API v10](https://discord.com/developers/docs/reference)

## Setup guide

> Follow these steps to create a discord developer account

Reviewer: ```suggestion
Follow these steps to create a discord developer account.
```

### Step 2: Make a New Discord Application

<img src="https://github.com/HussainLatiff/module-ballerinax-discord/blob/main/docs/setup/resources/make-new-app.png?raw=true">

Once in the discord developer portal is open, click on the `New Application` button as displayed above to start the process.

### Step 3: Name the Discord Application 

<img src="https://github.com/HussainLatiff/module-ballerinax-discord/blob/main/docs/setup/resources/create-app.png?raw=true">

1. Proceed by giving the Discord Application a name and click on the terms of service.

2. Finally complete the naming process by clicking on the `next` button

### Step 4: Obtain the Client ID and Client Secret

<img src="https://github.com/HussainLatiff/module-ballerinax-discord/blob/main/docs/setup/resources/obtain-client-id.png?raw=true">

Under the `OAuth2` section found on the left-sided list, locate the Client's Information as shown on the screen. To implement the functionalities provided by Discord's API, you will need the Client ID and Client Secret.

> 
> 

Reviewer: ```suggestion
```

<img src="https://github.com/HussainLatiff/module-ballerinax-discord/blob/main/docs/setup/resources/obtain-client-id.png?raw=true">

Under the `OAuth2` section found on the left-sided list, locate the Client's Information as shown on the screen. To implement the functionalities provided by Discord's API, you will need the Client ID and Client Secret.

## Quickstart

To use the `discord` connector in your Ballerina application, modify the `.bal` file as follows:

### Step 1: Import the module

Import the `discord` module.

```ballerina
import ballerinax/discord;
```

### Step 2: Instantiate a new connector

Create a `discord:ConnectionConfig` with the obtained OAuth2.0 Client Credentials and initialize the connector with it.

Apps must receive approval from users installing them to perform actions within Discord. To enable these functions, specific scopes must be defined. These scopes are outlined in the [OAuth2 Scopes documentation](https://discord.com/developers/docs/topics/oauth2#shared-resources-oauth2-scopes).

```ballerina
configurable string clientId = ?;
configurable string clientSecret = ?;
configurable string[] scopes = ?;

> discord:Client cl = check new({

Reviewer: ```suggestion
discord:Client discord = check new({
```

    auth: {
            clientId,
            clientSecret,
            scopes
        }
});
```

### Step 3: Invoke the connector operation
Now, utilize the available connector operations

> ### Return linked third-party accounts of the user

Reviewer: ```suggestion
```

```ballerina

>  var s = check cl->/users/\@me/connections();

Reviewer: ```suggestion
 var s = check discord->/users/\@me/connections();
```

Reviewer: Also, shall we come up with a more common use case for the quickstart, instead of having this? Maybe we can show how to send a message to a given discord channel.

Author: The reason we chose to go with this method is for a few reasons;
1. The scope that allows us to send messages along with the majority of scopes can only be used with discord's approval.

2. The other scopes that don't require approval like identify and email are outputting this error:
error: Unauthorized {"statusCode":401,"headers":{"alt-svc":["h3=":443"; ma=86400"],"cf-cache-status":["DYNAMIC"],"cf-ray":["89a4ccdaaf3f5134-CMB"],"content-length":["43"],"content-security-policy":["frame-ancestors 'none'; default-src 'none'"],"content-type":["application/json"],"date":["Thu, 27 Jun 2024 10:37:10 GMT"],"nel":["{"success_fraction":0,"report_to":"cf-nel","max_age":604800}"],"report-to":["{"endpoints":[{"url":"https:\/\/[a.nel.cloudflare.com](http://a.nel.cloudflare.com/)\/report\/v4?s=RsLp9HZVYtyMOdopqP628yT9NOgu5N4tEJA2Am8bFHZup%2Bk1hjTgethD3F2tgoDWIxWrJSgezyy1srOAa13qRCA8lXt2SWTtdIiMigqHj%2Bnalp1hfp40tHW4j2ld"}],"group":"cf-nel","max_age":604800}"],"server":["cloudflare"],"set-cookie":["__dcfduid=356d806e347111efb04f52c6d8280f9b; Expires=Tue, 26-Jun-2029 10:37:10 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/; SameSite=Lax","__sdcfduid=356d806e347111efb04f52c6d8280f9b5ff5a02c0a366e5c85761bc3520cbffc062f21ad716b6abe0703998acb68538c; Expires=Tue, 26-Jun-2029 10:37:10 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/; SameSite=Lax","__cfruid=e29dfc3f8738a18b6c2734aa1b8de839fc142d29-1719484630; path=/; domain=.[discord.com](http://discord.com/); HttpOnly; Secure; SameSite=None","_cfuvid=RzAQzdJtbkGOu7E0yMvG8xne0BpQUOaGXvSH8TbJ9XI-1719484630562-0.0.1.1-604800000; path=/; domain=.[discord.com](http://discord.com/); HttpOnly; Secure; SameSite=None"],"strict-transport-security":["max-age=31536000; includeSubDomains; preload"],"via":["1.1 google"],"x-content-type-options":["nosniff"],"x-http2-stream-id":["3"]},"body":{"message":"401: Unauthorized","code":0}}

We are currently trying to solve this issue, there are a few ideas we have to figuring this out and we will probably ask for help if it doesn't work.

3. The connections scope works and connects to the server and gives us a response.

Reviewer: @HussainLatiff I see... probably its okay to keep this example for the quickstart guide in that case. btw I've added a separate suggestion on the Ballerina code we have in the quickstart. Lets make sure to address them.
